src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java

```diff
@@ -180,28 +191,49 @@ private static void writeArray(JsonWriter writer, String name, Object value, Sch
 
     writer.name(name);
     writer.beginArray();
-    if (value instanceof Collection) {
-      for (Object element : (Collection) value) {
-        // big query does not allow null values in array items
-        if (element == null) {
-          throw new IllegalArgumentException(String.format("Field '%s' contains null values in its array, " +
-                                                             "which is not allowed by BigQuery.", name));
-        }
-        write(writer, name, true, element, componentSchema);
+
+    for (Object element : collection) {
+      // BigQuery does not allow null values in array items
+      if (element == null) {
+        throw new IllegalArgumentException(String.format("Field '%s' contains null values in its array, " +
+                                                           "which is not allowed by BigQuery.", name));
       }
-    } else {
-      for (int i = 0; i < Array.getLength(value); i++) {
-        // big query does not allow null values in array items
-        if (Array.get(value, i) == null) {
-          throw new IllegalArgumentException(String.format("Field '%s' contains null values in its array, " +
-                                                             "which is not allowed by BigQuery.", name));
+      if (element instanceof StructuredRecord) {
+        writer.beginObject();
```

Comment:
this block of code and the last part of the writeRecord() method are the same. Let's create a new method that is called in both places.

Reply:
Done.